### PR TITLE
Update Makefile and Dockerfile to support EKSConfig

### PR DESCRIPTION
**Pull request by rudoi**
It's probably a good idea to not merge this **through** the `eks-bootstrap` branch and instead open it as a follow-up once all the functionality is PR'd. There's a lot going on in here.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -240,19 +268,44 @@ generate-manifests: $(CONTROLLER_GEN) ## Generate manifests e.g. CRD, RBAC etc.
...
+		output:webhook:dir=./bootstrap/eks/config/webhook \
+		webhook
+
 ## --------------------------------------
 ## Docker
 ## --------------------------------------
 
 .PHONY: docker-build
-docker-build: ## Build the docker image for controller-manager
+docker-build:
```
Comment (h0tbird) [inline Makefile:288]: I don't have a default gcloud project so I had to set the `REGISTRY` to test this:
```
REGISTRY=gcr.io/foo make docker-build-core
```

**Comment by rudoi** [comment]
@h0tbird mind giving this another look? i just did some spicy rebasing and wanna make sure i didn't break the universe.

File: Makefile [hunk 2]
```diff
@@ -309,10 +327,20 @@ generate-go: $(CONTROLLER_GEN) $(CONVERSION_GEN) $(MOCKGEN) $(DEFAULTER_GEN) ##
...
+
+.PHONY: generate-go-eks-bootstrap
+generate-go-eks-bootstrap: $(CONTROLLER_GEN) $(CONVERSION_GEN)
+	$(CONTROLLER_GEN) \
+		paths=./bootstrap/eks/api/... \
+		paths=./bootstrap/eks/types/... \
+		object:headerFile=./hack/boilerplate/boilerplate.generatego.txt
 
 .PHONY: generate-manifests
-generate-manifests: $(CONTROLLER_GEN) ## Generate manifests e.g. CRD, RBAC etc.
```
Comment (h0tbird) [inline Makefile:315]: I don't think deleting all this line was your real intention:
```
make[1]: Nothing to be done for `generate-manifests'.
```

File: Makefile [hunk 3]
```diff
@@ -350,30 +403,39 @@ docker-build-%:
...
-	sed -i'' -e 's@image: .*@image: '"${MANIFEST_IMG}:$(MANIFEST_TAG)"'@' ./config/manager/manager_image_patch.yaml
-
+	sed -i'' -e 's@image: .*@image: '"${MANIFEST_IMG}:$(MANIFEST_TAG)"'@' $(TARGET_RESOURCE)
 
 .PHONY: set-manifest-pull-policy
 set-manifest-pull-policy:
-	$(info Updating kustomize pull policy file for manager resource)
-	sed -i'' -e 's@imagePullPolicy: .*@imagePullPolicy: '"$(PULL_POLICY)"'@' ./config/manager/manager_pull_policy.yaml
+	$(info Updating kustomize pull policy file for manager resources)
+	sed -i'' -e 's@imagePullPolicy: .*@imagePullPolicy: '"$(PULL_POLICY)"'@' $(TARGET_RESOURCE)
```
Comment (h0tbird) [inline Makefile:438]: I guess this is fine but it breaks direct invocations (which might not be used by anyone). The alternative is `set-core-manifest-image`, `set-core-manifest-pull-policy`....

Comment (rudoi) [reply Makefile:438]: FWIW I took this pattern from upstream CAPI: https://github.com/kubernetes-sigs/cluster-api/blob/4e07b0c23f6782fa5ca3729c3f9ed005a6f9a203/Makefile#L400-L408

File: Makefile (hunk 2 above)
Comment (rudoi) [reply Makefile:315]: ah definitely not! will fix

**Pull request merged**